extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java

```diff
@@ -168,7 +168,11 @@ private DeploymentResultBuildItem deploy(DeploymentTargetEntry deploymentTarget,
         try (FileInputStream fis = new FileInputStream(manifest)) {
             KubernetesList list = Serialization.unmarshalAsList(fis);
             distinct(list.getItems()).forEach(i -> {
-                client.resource(i).inNamespace(namespace).createOrReplace();
+                if ("knative".equals(deploymentTarget.getName().toLowerCase())) {
```

Comment:
Very minor but is there a reason why you use a string here rather than using the `KNATIVE` constant as below?

Reply:
Indeed, @iocanel can you fix this so we can merge?